madmom/utils/__init__.py

```diff
@@ -419,6 +426,97 @@ def quantize_events(events, fps, length=None, shift=None):
     return quantized
 
 
+def quantize_notes(notes, fps, length=None, num_notes=None, shift=None,
+                   offset=None, velocity=None):
+    """
+    Quantize the notes with the given resolution.
+
+    Parameters
+    ----------
+    notes : 2D numpy array
+        Notes to be quantized.
+    fps : float
+        Quantize with `fps` frames per second.
+    length : int, optional
+        Length of the returned array. If 'None', the length will be set
+        according to the latest sounding note.
+    num_notes : int, optional
+        Number of notes of the returned array. If 'None', the number of notes
+        will be based on the highest note number in the `notes` array. All note
+        numbers equal or greater this value will not be included in the
+        returned quantized array.
+    shift : float, optional
```

Comment:
I don't think `shift` or `offset` should be part of this function.

1) It is very easy to implement both outside of the function
2) It makes the function signature unnecessarily complex

Reply:
True, at least for `offset`. On the other hand, since `shift` can be a float and rounding is used inside the function, moving this outside the function may introduces more errors.

Reply:
IMHO the following is not more error-prone:
```python
notes[:, 0] += 2.3
qnotes = quantize_notes(notes)
```
than this:
```python
qnotes = quantize_notes(notes, shift=2.3)
```
If there is a common use case for the shift, then the second option is fine, but if it is a special case, I vote for the first one, because I think a function should do only one thing, if possible. That said, I don't have a strong opinion on this specific case.

Reply:
You are right, I somehow only considered applying the shift after calling the function... I will remove it (and deprecate the shift in `quantize_events`).